hs_core/tasks.py

```diff
@@ -158,6 +168,69 @@ def nightly_periodic_task_check():
         file.write(timezone.now().strftime('%m/%d/%y %H:%M:%S'))
 
 
+@celery_app.task(ignore_result=True, base=HydroshareTask)
+def nightly_repair_resource_files():
+    """
+    Run repair_resource management command on resources updated in the last day
+    """
+    from hs_core.management.utils import check_time, repair_resource
+    start_time = time.time()
+    cuttoff_time = timezone.now() - timedelta(days=1)
+    recently_updated_resources = [res for res in BaseResource.objects.all() if res.last_updated >= cuttoff_time]
+    repaired_resources = []
+    try:
+        for res in recently_updated_resources:
+            check_time(start_time, settings.NIGHTLY_RESOURCE_REPAIR_DURATION)
+            is_corrupt = repair_resource(res, logger)
+            if is_corrupt:
+                repaired_resources.append(res)
+
+        # spend any remaining time fixing resources that haven't been checked
+        # followed by those previously checked, prioritizing the oldest checked date
+        recently_updated_rids = [res.short_id for res in recently_updated_resources]
+        not_recently_updated = BaseResource.objects.exclude(short_id__in=recently_updated_rids) \
+            .order_by(F('files_checked').asc(nulls_first=True))
+        for res in not_recently_updated:
+            check_time(start_time, settings.NIGHTLY_RESOURCE_REPAIR_DURATION)
+            is_corrupt = repair_resource(res, logger)
+            if is_corrupt:
+                repaired_resources.append(res)
+    except TimeoutError:
```

Comment:
If we get a `TimeoutError`, we don't need to process any of the remaining resources?

Reply:
My thought here is that we only allot 1hr per night to fixing resources. If we run out of time, add any remaining to the queue for the next night. I did this because I think if we don't limit the resource repair, it could literally run on the worker for almost a week. This way we gradually chip away at the backlog every night

Reply:
My question was confusing. If we run out of time then  I am suggesting we should be getting out of the for loop for resources to be repaired as we won't be able to repair any of the remaining resources.

Reply:
Actually wait, I believe that `check_time` raises a `TimeoutError` that will terminate the `for res in not_recently_updated:` loop. So the remaining resources would not be repaired.

This is also true for the `for res in recently_updated_resources:` loop.

Reply:
Yeah I see it now.